memote/suite/cli/reports.py

```diff
@@ -153,11 +153,71 @@ def history(location, filename, custom_config):
+                   "All other tests are skipped. This option can be used "
+                   "multiple times and takes precedence over '--skip'.")
+@click.option("--skip", type=str, multiple=True, metavar="TEST",
+              help="The name of a test or test module to be skipped. This "
+                   "option can be used multiple times.")
+@click.option("--solver", type=click.Choice(["cplex", "glpk", "gurobi"]),
+              default="glpk", show_default=True,
+              help="Set the solver to be used.")
+@click.option("--experimental", type=click.Path(exists=True, dir_okay=False),
+              default=None, callback=callbacks.validate_experimental,
+              help="Define additional tests using experimental data.")
+@click.option("--custom-tests", type=click.Path(exists=True, file_okay=False),
+              multiple=True,
+              help="A path to a directory containing custom test "
+                   "modules. Please refer to the documentation for more "
+                   "information on how to write custom tests. May be "
+                   "specified multiple times.")
+@click.option("--custom-config", type=click.Path(exists=True, dir_okay=False),
+              multiple=True,
+              help="A path to a report configuration file that will be merged "
+                   "into the default configuration. It's primary use is to "
+                   "configure the placement and scoring of custom tests but "
+                   "it can also alter the default behavior. Please refer to "
+                   "the documentation for the expected YAML format used. This "
+                   "option can be specified multiple times.")
+def diff(models, filename, pytest_args, exclusive, skip, solver,
+         experimental, custom_tests, custom_config):
+    """
+    Take a snapshot of all the supplied models and generate a diff report.
+
+    MODELS: List of paths to model files.
+    """
+    if not any(a.startswith("--tb") for a in pytest_args):
+        pytest_args = ["--tb", "no"] + pytest_args
+    # Add further directories to search for tests.
+    pytest_args.extend(custom_tests)
+    config = ReportConfiguration.load()
+    # Update the default test configuration with custom ones (if any).
+    for custom in custom_config:
+        config.merge(ReportConfiguration.load(custom))
+    # Build the diff report specific data structure
+    diff_results = dict()
+    for model_path in models:
+        try:
+            model_filename = model_path.split('/')[-1].split('.')[0]
+            diff_results.setdefault(model_filename, dict())
+            model = callbacks._load_model(model_path)
+            model.solver = solver
+            _, diff_results[model_filename] = api.test_model(
+                model, results=True, pytest_args=pytest_args,
+                skip=skip, exclusive=exclusive, experimental=experimental)
+        except Exception as e:
```

Comment:
This should be handled in the `callback._load_model` then. Let's find a way to aggregate and display those warnings.

Reply:
So should I just move the `try/except` statement over into `_load_model` or do we want to look at this in a separate PR maybe?